packages/api-plugin-fulfillment/migrations/4.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @summary Performs migration up from previous data version
+ * @param {Object} context Migration context
+ * @param {Object} context.db MongoDB `Db` instance
+ * @param {Function} context.progress A function to report progress, takes percent
+ *   number as argument.
+ * @return {undefined}
+ */
+async function up({ db, progress }) {
+  progress(0);
+  const affectedGroups = [
+    "owner",
+    "shop manager"
+  ];
+
+  const newShopPermissions = [
+    "reaction:legacy:fulfillmentTypes/update:settings",
+    "reaction:legacy:fulfillmentRestrictions/create",
+    "reaction:legacy:fulfillmentRestrictions/delete",
+    "reaction:legacy:fulfillmentRestrictions/read",
+    "reaction:legacy:fulfillmentRestrictions/update",
+    "reaction:legacy:fulfillmentTypes/create",
+    "reaction:legacy:fulfillmentTypes/delete",
+    "reaction:legacy:fulfillmentTypes/read",
+    "reaction:legacy:fulfillmentTypes/update",
+    "reaction:legacy:fulfillmentMethods/create",
+    "reaction:legacy:fulfillmentMethods/delete",
+    "reaction:legacy:fulfillmentMethods/read",
+    "reaction:legacy:fulfillmentMethods/update"
+  ];
+
+  await db.collection("Groups").updateMany({
+    slug: { $in: affectedGroups }
+  }, {
+    $addToSet: { permissions: { $each: newShopPermissions } }
+  });
+
+  progress(100);
+}
+
+export default {
+  down: "impossible",
```

Comment:
Seems like down migration would actually be possible with similar code except `$pullAll` rather than `$addToSet`.

Reply:
Yes, I am adding the down migration

Reply:
@aldeed @zenweasel 
**Note:**
Additionally, I removed the below entry since it would be better to include this entry in the PR #6543 
`"reaction:legacy:fulfillmentTypes/update:settings",`

Although this would require a separate migration script for only this entry, the change actually belong to that PR.

Reply:
As discussed, I would be re-introducing the removed entry back to this PR as it is preferred to have a single migration

Reply:
Done